### PR TITLE
Add auto deployment service

### LINKED ISSUE #603: Add automatically generated wireguard AllowedIP list to credentialspage
Developer request @mguidon

### PR DESCRIPTION
**Pull request by sanderegg**
Moved deployment-agent from internal gitlab to github.
closes #602

Bonuses:
- [x] fixes #603 by re-ordering the calls that were generating a syntax error.
- [x] added CI in repository
- [x] added basic tests
- [x] added CD in repository

### CODE REVIEW

File: ops/travis/unit-testing/deployment-agent [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -e
+
+FOLDER_CHECKS="deployment-agent\/"
```
Comment (oetiker) [inline ops/travis/unit-testing/deployment-agent]: a forward slash does not need escaping

File: ops/travis/unit-testing/deployment-agent [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+before_install() {
+    if [ -n "$(bash ops/travis/helpers/test_for_changes deployment-agent)" ];
```
Comment (oetiker) [inline ops/travis/unit-testing/deployment-agent]: using the aproach I added in my simcore PR seems nicer to me :)

File: services/deployment-agent/src/simcore_service_deployment_agent/subtask.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    async def init(self):
+        pass
+    
+    @abstractmethod
+    async def check_for_changes(self):
+        pass
+    
+    @abstractmethod
+    async def cleanup(self):
+        pass
```
Comment (oetiker) [inline services/deployment-agent/src/simcore_service_deployment_agent/subtask.py]: last newline missing ... there is a code setting to check for that

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+info:
+	@echo '+ VCS_* '
+	@echo '  - ULR                : ${VCS_URL}'
+	@echo '  - REF                : ${VCS_REF}'
+	@echo '  - (STATUS)REF_CLIENT : (${VCS_STATUS_CLIENT})'
+	@echo '+ BUILD_DATE           : ${BUILD_DATE}'
+	@echo '+ VERSION              : ${VERSION}'
+	@echo '+ DOCKER_REGISTRY      : ${DOCKER_REGISTRY}'
+	@echo '+ SERVICES_VERSION     : ${SERVICES_VERSION}'
+	@echo '+ PY_FILES             : $(shell echo $(PY_FILES) | wc -w) files'
```
Comment (pcrespov) [inline Makefile]: PY_FILES, SERVICES_VERSION , ... do not exist

File: services/deployment-agent/docker/healthcheck_curl_host.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/bin/python
+
+import json
+import os
+import sys
+from urllib.request import urlopen
+
+BOOTS_WITH_DEBUGGER = "2"
```
Comment (pcrespov) [inline services/deployment-agent/docker/healthcheck_curl_host.py]: the environ created in Dockerfile is called ``SC_BOOT_MODE``

File: services/deployment-agent/requirements/prod.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+# paths relative to location of setup.py
+.
+#./client-sdk
+git+https://github.com/ITISFoundation/osparc-simcore.git#subdirectory=packages/service-library
```
Comment (pcrespov) [inline services/deployment-agent/requirements/prod.txt]: interesting ... now we might think of having release versions of service-library?

File: services/deployment-agent/docker/healthcheck_curl_host.py (hunk 5 above)
Comment (sanderegg) [reply services/deployment-agent/docker/healthcheck_curl_host.py]: thx for that one. fixed.

File: services/deployment-agent/requirements/prod.txt (hunk 6 above)
Comment (sanderegg) [reply services/deployment-agent/requirements/prod.txt]: we'll probably have to think of it as it is used everywhere..

File: Makefile (hunk 4 above)
Comment (sanderegg) [reply Makefile]: fixed.

File: services/deployment-agent/src/simcore_service_deployment_agent/subtask.py (hunk 3 above)
Comment (sanderegg) [reply services/deployment-agent/src/simcore_service_deployment_agent/subtask.py]: done

File: ops/travis/unit-testing/deployment-agent (hunk 2 above)
Comment (sanderegg) [reply ops/travis/unit-testing/deployment-agent]: done

File: ops/travis/unit-testing/deployment-agent (hunk 1 above)
Comment (sanderegg) [reply ops/travis/unit-testing/deployment-agent]: done

**Pull request merged**